medusa/purge.py

```diff
@@ -62,6 +68,17 @@ def main(config, max_backup_age=0, max_backup_count=0):
         sys.exit(1)
 
 
+def backups_to_purge_by_completion(backups):
+    complete, incomplete = [], []
+    for backup in backups:
+        complete.append(backup) if backup.finished is not None else incomplete.append(backup)
```

Comment:
**Issue:** This doesn't work the expected way I think. Here, the completeness of the backup is checked for the local node only (we're dealing with a NodeBackup, not a ClusterBackup).
Hence, any incomplete ClusterBackup would still be considered a complete backup for the purge process.
Remember that the purge is done locally for a node, it's not a global operation and has to run on each node.